### DISCUSSION TITLE
Can I use MediatoR in .net 6 winui3 project

### DISCUSSION THREAD

**Discussion by hkamala** [LuckyPennySoftware/MediatR#909] [General]
I am new to winui3 desktop application, I am trying to implement CQRS in my winui3 project. but Got to know the dependency injection support asp.net core so, suggest which nuget package I have to use for .net 6

**Comment by jbogard** [comment]
The latest one supports .NET 6. And 7, and 8, and .NET 4.7.2.

**Comment by hkamala** [comment]
I have 2 different solutions in the same project . in which one solution contain 
1. CQRS implementation
2. WinUI 3 application
in app.xaml.cs  of winui3 project I am trying to add the dependency injection
 
services.AddMediatR(typeof(AuthorizedUserRepository).Assembly);

but I am getting 
InnerException	{"Unable to resolve service for type 'System.Data.IDbTransaction' 
while attempting to activate 'RTMS.Repositories.AuthorizedUserRepository'."}
	System.Exception {System.InvalidOperationException}

could you please help in resolving this error.

**Reply by hkamala** [reply]
System.InvalidOperationException: 'Error constructing handler for request of type MediatR.IRequestHandler`2[RTMS.Services.CQRS.GetAuthorizedUserQuery,FluentResults.Result`1[System.Collections.ObjectModel.ObservableCollection`1[RTMS.Services.CQRS.GetAuthorizedUserQueryResponse]]]. Register your handlers with the container. See the samples in GitHub for examples.'


this is the error message

**Reply by hkamala** [reply]
In my UI project we have these NuGet packages

1.MediatR.Extensions.Microsoft.DependencyInjection
2.Microsoft.Extensions.DependencyInjection
